cleverhans/experimental/certification/dual_formulation.py

```diff
@@ -209,45 +213,76 @@ def project_dual(self):
     """Function that projects the input dual variables onto the feasible set.
 
     Returns:
-      projected_dual: Feasible dual solution corresponding to current dual
       projected_certificate: Objective value of feasible dual
     """
-    # TODO: consider whether we can use shallow copy of the lists without
```

Comment:
The MNIST certification example still works with this changed to the shallow copy. Haven't run all the tests though.